src/app/utilities/AnnotateUtilities.ts

```diff
@@ -2,70 +2,101 @@
-}) => {
-    const { annotationTarget, superClass } = params;
-
-    /* Base variables */
-    const pathArray: string[] = ['$'];
-    const parents = jp.parse(annotationTarget.jsonPath).slice(1, -1);
-    const parentClasses: ParentClass[] = [];
-
-    /* For each parant, find and extract its class information */
-    parents.forEach((parent, index) => {
-        /* Filter path from schemas */
-        let parentName = parent.expression.value.replace('ods:', '').replace('dwc:', '').replace('dcterms:', '');
-
-        /* Add to pathArray */
-        pathArray.push(parent.expression.value);
-
-        if (parentName.includes('has') || parentName[0] === parentName[0].toUpperCase()) {
-            /* If a parent class is empty, it needs to be targeted, if it is not completely empty, check for indexes */
-            const parentNodes = jp.nodes(superClass, jp.stringify(pathArray).replaceAll('][', ']..[').replaceAll('"', "'"));
-
-            if (!parentNodes.length) {
-                parentClasses.push({
-                    jsonPath: jp.stringify(pathArray).replaceAll('"', "'"),
-                    name: MakeJsonPathReadableString(jp.stringify(pathArray)),
-                    ...(index > 0 && { parentName: pathArray[index] }),
-                    present: false
-                });
-            }
-
-            parentNodes.forEach((parentNode, parentIndex) => {
-                parentClasses.push({
-                    jsonPath: jp.stringify(parentNode.path).replaceAll('"', "'"),
-                    name: MakeJsonPathReadableString(jp.stringify(parentNode.path)),
-                    ...(index > 0 && { parentName: MakeJsonPathReadableString(pathArray[index]) }),
-                    present: (Array.isArray(parentNode.value) && !!parentNode.value.length) || (!Array.isArray(parentNode.value) && typeof (parentNode.value) === 'object'),
-                    ...(parentNode.value.length && { options: parentNode.value.length }),
-                    ...(parentClasses[parentIndex]?.options && { dependent: true })
+const FormatFieldNameFromJsonPath = (jsonPath: string) => {
+    return jsonPath.replaceAll('.', '_').replaceAll('][', '_').replaceAll('[', '').replaceAll(']', '');
+};
+
+/**
+ * Function to generate and return a dictionary containing all classes and terms of the schema adhering to the provided schema name; and their values
+ * @param jsonPath The JSON path that leads to the schema to be used
+ * @returns The annotation form field properties and their associated form values
+ */
+const GenerateAnnotationFormFieldProperties = async (jsonPath: string, superClass: DigitalSpecimen | DigitalMedia | Dict) => {
+    const annotationFormFieldProperties: {
+        [propertyName: string]: AnnotationFormProperty
+    } = {};
+    const formValues: Dict = {};
+
+    /* Extract main schema */
+    await ExtractFromSchema(jsonPath).then(async (schema) => {
+        await ExtractClassesAndTermsFromSchema(schema).then(({
```

Comment:
Enforced better typing in one of the upcoming pull requests (made a bunch of changes and am afraid for merge conflicts as it works now). It can still be undefined, but it then replaced with an empty dict (object)